File: javascript/recursion/Tak.js
function tak(x, y, z) {
    if (y < x) {
        return tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y))
    } else {
        return z
    }
}

const startTimeMs = new Date().getTime()
      
console.log(tak(30, 22, 12))

const endTimeMs = new Date().getTime()
const durationMs = endTimeMs - startTimeMs
console.log(`Execution time: ${durationMs}ms`)

File: perl/recursion/Tak.pl
use strict;
use warnings;
use Time::HiRes qw(time);

sub tak {
    my ($x, $y, $z) = @_;
    return $y < $x
        ? tak(tak($x - 1, $y, $z), tak($y - 1, $z, $x), tak($z - 1, $x, $y))
        : $z;
}

sub main {
    my $startTimeMs = int(time * 1000);

    print tak(30, 22, 12), "\n";

    my $endTimeMs = int(time * 1000);
    my $durationMs = $endTimeMs - $startTimeMs;

    print "Execution time: " . $durationMs . "ms\n";
}

main();
